File: src/main/java/com/thesilentnights/easylogin/utils/LogUtil.java
package com.thesilentnights.easylogin.utils;

import com.thesilentnights.easylogin.repo.CommonStaticRepo;

public class LogUtil {
    public static void logInfo(Class<?> clazz, String message) {
        CommonStaticRepo.log.info("{} : {}", clazz.getSimpleName(), message);
    }


    public static void logError(Class<?> clazz, String message, Exception error) {
        CommonStaticRepo.log.error("{} : {}", clazz.getSimpleName(), message, error);
    }
}

File: src/main/kotlin/com/thesilentnights/easylogin/utils/LogUtil.kt
package com.thesilentnights.easylogin.utils

import com.thesilentnights.easylogin.repo.CommonStaticRepo
import kotlin.reflect.KClass

fun logInfo(clazz: KClass<*>, message: String) {
    CommonStaticRepo.log.info(clazz.simpleName + " : ", message)
}

fun logInfo(clazz: Class<*>, message: String) {
    CommonStaticRepo.log.info(clazz.simpleName + " : " + message)
}

fun logError(clazz: KClass<*>, message: String, error: Throwable) {
    CommonStaticRepo.log.error(clazz.simpleName + " : " + message, error)
}

fun logError(clazz: Class<*>, message: String,error: Throwable) {
    CommonStaticRepo.log.error(clazz.simpleName + " : " + message)
}
